### PR TITLE
Refresh requirements with pip-compile

### PR DESCRIPTION
**Pull request by richardsheridan**
It looks like several transitive requirements have changed and that dependabot has been dropping the ball on some updates.

### CODE REVIEW

**Comment by richardsheridan** [comment]
Looks like I was a bit optimistic 😅  Looks like the main error is Ipython and building docs. I saw that same error building docs locally and I think it went away when I did `pip install -e .` as for IPython, i'll try just reverting that version bump.

**Comment by richardsheridan** [comment]
> I saw that same error building docs locally and I think it went away when I did `pip install -e .` 

Even weirder, it seems that it was not  `pip install -e .`  that fixed it locally, but merely running the sphinx build twice in the same env. No idea what that means! 🤷

**Comment by pquentin** [comment]
Can you please revert the Sphinx bump too? It's known to break our docs

**Comment by richardsheridan** [comment]
Hm! Sphinx 5.0.0 is apparently not the direct cause of the RTD problems. I will poke around with some of the other dependency versions this weekend.

**Comment by pquentin** [comment]
> Hm! Sphinx 5.0.0 is apparently not the direct cause of the RTD problems. 

Yeah, the last version that works is 3.3.1: https://github.com/python-trio/trio/pull/2338

File: docs-requirements.in [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # RTD is currently installing 1.5.3, which has a bug in :lineno-match:
-sphinx >= 1.7.0
+# sphinx-3.4 causes wanings about some trio._abc classes: GH#2338
```
Comment (richardsheridan) [inline docs-requirements.in:2]: Note to self, typo "warnings"

**Comment by pquentin** [comment]
(Unassigned me since this it does not look ready. Please ping me if it does become ready! Thanks)

**Review by pquentin — APPROVED** [review]
Thanks!

**Pull request merged**